2023/10/24/main.go

```diff
@@ -0,0 +1,13 @@
+// Daily Joel 2023-10-24
+//
+// Stacking defers.
+package main
+
+import "fmt"
+
+func main() {
+	letters := []string{"\n", "l", "e", "o", "J"}
+	for _, l := range letters {
+		defer fmt.Print(l)
+	}
+}
```

Comment:
Each time the print function is called, defer will delay the print last.

Reply:
Deferred functions are stored as a stack. Each new deferral will be pushed on top of the stack, and when the surrounding function returns the stack will be executed _from the top_ in a last-in-first-out approach. This is why we need to stack the print functions in reversed order to get `Joel\n` in the end

There's a nice visual representation here
https://en.wikipedia.org/wiki/Stack_(abstract_data_type)